File: javascript/src/televentas/usuario.js
export class Usuario {
  constructor(idUsuario, nombre, correo, activo) {
    if (new.target === Usuario) {
      throw new TypeError("No se puede instanciar la clase abstracta Usuario");
    }
    this.idUsuario = idUsuario;
    this.nombre = nombre;
    this.correo = correo;
    this.activo = activo;
  }

  autenticar() {
    return this.activo;
  }

  obtenerDatos() {
    return `Usuario(id=${this.idUsuario}, nombre=${this.nombre}, correo=${this.correo})`;
  }

  getIdUsuario() {
    return this.idUsuario;
  }

  getNombre() {
    return this.nombre;
  }

  getCorreo() {
    return this.correo;
  }

  isActivo() {
    return this.activo;
  }
}

File: java/src/televentas/Usuario.java
package televentas;

public abstract class Usuario {

    private final int idUsuario;
    private final String nombre;
    private final String correo;
    private final boolean activo;

    protected Usuario(int idUsuario, String nombre, String correo, boolean activo) {
        this.idUsuario = idUsuario;
        this.nombre = nombre;
        this.correo = correo;
        this.activo = activo;
    }

    public boolean autenticar() {
        return activo;
    }

    public String obtenerDatos() {
        return String.format(
                "Usuario(id=%d, nombre=%s, correo=%s)",
                idUsuario, nombre, correo
        );
    }

    public int getIdUsuario() {
        return idUsuario;
    }

    public String getNombre() {
        return nombre;
    }

    public String getCorreo() {
        return correo;
    }

    public boolean isActivo() {
        return activo;
    }
}
